MariaDB/server
# MDEV-27394 zero out addr.un.sun_path string so log message on socket is correct

- [x] *The Jira issue number for this PR is: [MDEV-27394](https://jira.mariadb.org/browse/MDEV-27394)

## Description
If this char[] is not zeroed out, inaccurate log message described in [MDEV-27394](https://jira.mariadb.org/browse/MDEV-27394) will show up, due to garbage data happens to be in that memory location.

## How can this PR be tested?
This PR just corrects the previously messed up log message. Like [PR for MDEV-17093](https://github.com/MariaDB/server/pull/1979), a test case may not be necessary.

To reproduce and verify:
1. Run `systemd-socket-activate -l @abstractlong1 -l @abstracts2 -l @abstractverylog  mysqld  --no-defaults --bootstrap` in one terminal session;
2. On a different terminal session, run `mysql -S @abstracts2`;
3. Observe the output message difference. `mysqld` without this fix will display incorrect socket name.

## Basing the PR against the correct MariaDB version
